### DISCUSSION TITLE
Error Message: MuPDF error: format error: cmsOpenProfileFromMem failed

### DISCUSSION THREAD

**Discussion by isaac-peter** [pymupdf/PyMuPDF#3572] [Looking for help] [answered]
I am processing over 35,000 pdfs, and I continually get error message: "MuPDF error: format error: cmsOpenProfileFromMem failed".

I cannot find any reference to this error in the documentation and Google is not much help. 

Is anyone familiar with A) What this error is in reference to and B) the ramifications of this error in terms of data extraction?

Thx in advance!

**Comment by JorjMcKie** [comment, marked as answer]
This message is issued when unsuccessfully trying to interpret ICC color profiles. Presumably there exist erroneous profiles in some PDFs.
The message is issued from our base library, MuPDF so you may want to ask for more background in the public [MuPDF Discord](https://discord.gg/V75rKf6p) channel.
As a quick circumvention, why don't you wrap the processing in try/except clauses?
Another option might be to switch off ICC support by executing `pymupdf.TOOLS.set_icc(False)`.

**Reply by isaac-peter** [reply]
Great! Thank you!

**Comment by benoit74** [comment]
The `pymupdf.TOOLS.set_icc(False)` "trick" did not worked for 

I got following advice on Discord which worked well for my use case (I don't mind at all about PDF images)

> you can always suppress all MuPDF error messages via pymupdf.TOOLS.mupdf_display_errors(False). They will still be available in the internally maintained string pymupdf.TOOLS.mupdf_warnings() together with any warnings.

**Reply by JorjMcKie** [reply]
Thanks for reporting this!
BTW I saw you also are a C# programmer. Did you notice [mupdf.net](https://github.com/ArtifexSoftware/MuPDF.NET), the C# port of PyMuPDF?

**Reply by benoit74** [reply]
Yep, thanks for noticing it. I'm now doing mostly only Python, so I'm probably not going to use mupdf.net in the near future.

**Reply by dragongu** [reply]
@benoit74 HI，I also encountered a similar problem, that is to say, does this error not affect the function?
